plugins/jobs/src/js/JobCreateEditFormModal.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from "react";
+
+import { BehaviorSubject } from "rxjs/BehaviorSubject";
+import { combineLatest } from "rxjs/observable/combineLatest";
+
+import { componentFromStream } from "data-service";
+
+import { createJob, JobData, updateJob } from "#SRC/js/events/MetronomeClient";
+import Job from "#SRC/js/structs/Job";
+
+import JobFormModal, { ErrorMessage } from "./components/JobFormModal";
+
+interface JobCreateEditFormModalProps {
+  isEdit?: boolean;
+  job?: Job;
+  open: boolean;
+  onClose: () => void;
+}
+
+type JobFormModalInput = [JobCreateEditFormModalProps, ErrorMessage];
+
+const JobCreateEditFormModal = componentFromStream<JobCreateEditFormModalProps>(
+  props$ => {
+    const errorMessage$ = new BehaviorSubject<ErrorMessage>(null);
+
+    function handleErrorMessageChange(errorMessage: ErrorMessage) {
+      errorMessage$.next(errorMessage);
+    }
+
+    function handleSubmit(isEdit: boolean, jobSpec: JobData, jobId: string) {
+      const request$ = isEdit ? updateJob(jobId, jobSpec) : createJob(jobSpec);
+
+      combineLatest(props$, request$)
+        .take(1) // unsubscribe after the first one to break ∞ react update loop
```

Comment:
There's also `first()` we could use. We need this because `props.onClose()` changes state of the parent component which pushes new props down the `props$` stream ...which triggers `props.onClose()` :)

Reply:
in the following PR